src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,59 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T, V> {
+    private T firstPair;
+    private V secondPair;
+
+    private Pair(T firstPair, V secondPair) {
+        this.firstPair = firstPair;
+        this.secondPair = secondPair;
+    }
+
+    public T getFirst() {
+        return firstPair;
+    }
+
+    public void setFirst(T firstPair) {
+        this.firstPair = firstPair;
+    }
+
+    public V getSecond() {
+        return secondPair;
+    }
+
+    public void setSecond(V secondPair) {
+        this.secondPair = secondPair;
+    }
+
+    public static <T, V> Pair<T, V> of(T firstPair, V secondPair) {
+        return new Pair<>(firstPair, secondPair);
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+
+        Pair<?, ?> pair = (Pair<?, ?>) object;
+
+        if (!Objects.equals(firstPair, pair.firstPair)) {
+            return false;
+        }
+        return Objects.equals(secondPair, pair.secondPair);
```

Comment:
please see checklist

Reply:
Reworked... ^__^
I think it, should be fine, IDE even wants to replace it with Objects.equals().

Reply:
later we will use Objects.equals() but first we should learn how to write it ourselves